consensus/spos/commonSubround/subroundStartRound.go

```diff
@@ -48,6 +51,7 @@ func NewSubroundStartRound(
 		executeStoredMessages,
 		broadcastUnnotarisedBlocks,
 		statusHandler.NewNilStatusHandler(),
+		nil,
```

Comment:
why are you giving nil here ? You can do the same thing as NilStatusHandler and call it NilIndexer (Null Object Design Pattern)

Reply:
I set this parameter with nil value because the majority of the nodes will not use indexer because only one observer per shard will send data to the elastic search.

Reply:
ok, got it. but please make sure that everywhere you try to access the indexer you check it for nil

Reply:
use the design pattern suggested by bogdan. Instantiate it in the constructor with NilIndexer.

Reply:
Done.

Reply:
Done.